hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockStreamManagerImpl.java

```diff
@@ -228,8 +213,7 @@ public void startRound(@NonNull final Round round, @NonNull final State state) {
         // Writer will be null when beginning a new block
         if (writer == null) {
             writer = writerSupplier.get();
-            // This iterator is never empty; c.f. DefaultTransactionHandler#handleConsensusRound()
-            blockTimestamp = round.iterator().next().getConsensusTimestamp();
+            blockTimestamp = round.getConsensusTimestamp();
```

Comment:
How does this line cover the four new scenarios added to `block_header.proto` above?

Reply:
This logic is encapsulated in the round's construction, not in the determination of which timestamp execution assigns to the block 👍